packages/core/src/components/forms/numericInput.tsx

```diff
@@ -0,0 +1,407 @@
+
+import * as classNames from "classnames";
+import * as PureRender from "pure-render-decorator";
+import * as React from "react";
+
+import { AbstractComponent, Classes, Keys } from "../../common";
+import * as Errors from "../../common/errors";
+import { Position } from "../../common/position";
+import { HTMLInputProps, IIntentProps, IProps, removeNonHTMLProps } from "../../common/props";
+import * as Utils from "../../common/utils";
+
+import { Button } from "../button/buttons";
+import { InputGroup } from "./inputGroup";
+
+export interface INumericInputProps extends IIntentProps, IProps {
+
+    /**
+     * The position of the buttons with respect to the input field
+     * @default Position.RIGHT
+     */
+    buttonPosition?: Position.LEFT | Position.RIGHT | "none";
+
+    /**
+     * Whether the input is in a non-interactive state
+     * @default false
+     */
+    disabled?: boolean;
+
+    /** The name of icon (the part after `pt-icon-`) to render on left side of input */
+    leftIconName?: string;
+
+    /** The placeholder text in the absence of any value */
+    placeholder?: string;
+
+    /**
+     * The increment between successive values when `shift` is held
+     * @default 10
+     */
+    majorStepSize?: number;
+
+    /** The maximum value of the input */
+    max?: number;
+
+    /** The minimum value of the input */
+    min?: number;
+
+    /**
+     * The increment between successive values when `alt` is held
+     * @default 0.1
+     */
+    minorStepSize?: number;
+
+    /**
+     * The increment between successive values when no modifier keys are held
+     * @default 1
+     */
+    stepSize?: number;
+
+    /** The value to display in the input field */
+    value?: number | string;
+
+    /** The callback invoked when `enter` is pressed and when the field loses focus */
+    onConfirm?(value: string): void;
+
+    /** The callback invoked when the value changes */
+    onUpdate?(value: string): void;
+}
+
+export interface INumericInputState {
+    isInputGroupFocused?: boolean;
+    isButtonGroupFocused?: boolean;
+    shouldSelectAfterUpdate?: boolean;
+    value?: string;
+}
+
+@PureRender
+export class NumericInput extends AbstractComponent<HTMLInputProps & INumericInputProps, INumericInputState> {
+    public static displayName = "Blueprint.NumericInput";
+
+    public static defaultProps: INumericInputProps = {
+        buttonPosition: Position.RIGHT,
+        majorStepSize: 10,
+        minorStepSize: 0.1,
+        stepSize: 1,
+        value: NumericInput.VALUE_EMPTY,
+    };
+
+    private static DECREMENT_KEY = "decrement";
+    private static INCREMENT_KEY = "increment";
+    private static DECREMENT_ICON_NAME = "chevron-down";
+    private static INCREMENT_ICON_NAME = "chevron-up";
+    private static VALUE_EMPTY = "";
+    private static VALUE_ZERO = "0";
+
+    private inputElement: HTMLInputElement;
+
+    public constructor(props?: HTMLInputProps & INumericInputProps) {
+        super(props);
```

Comment:
btw, I was wrong about this... https://github.com/palantir/tslint-react/issues/54

but the whole codebase now has this problem which should be addressed in a separate PR